cli/pcluster/config/validators.py

```diff
@@ -572,6 +572,24 @@ def url_validator(param_key, param_value, pcluster_config):
     return errors, warnings
 
 
+def s3_bucket_validator(param_key, param_value, pcluster_config):
+    errors = []
+    warnings = []
+
+    if urlparse(param_value).scheme == "s3":
+        try:
+            bucket = param_value.split("/")[2]
+            boto3.client("s3").head_bucket(Bucket=bucket)
```

Comment:
this validation does not guarantee that the nodes of the cluster have access to the bucket (because the user policy is different that the role assumed by cluster nodes) and also enforces the user that creates the cluster to have head_bucket access to the s3 bucket itself.